metpy/calc/indices.py

```diff
@@ -0,0 +1,51 @@
+# Copyright (c) 2008-2015 MetPy Developers.
+# Distributed under the terms of the BSD 3-Clause License.
+# SPDX-License-Identifier: BSD-3-Clause
+"""Contains calculation of various derived indicies."""
+
+from ..calc import mixing_ratio, saturation_vapor_pressure
+from ..calc.tools import get_layer
+from ..constants import g, rho_l
+from ..package_tools import Exporter
+from ..units import units
+
+exporter = Exporter(globals())
+
+
+def precipitable_water(dewpt, p, top=400 * units('hPa')):
+    # Implementation of integral calculation based partly on SRH integral
+    # approximation, as found in Markowski and Richardson (2013) or in
+    # SharpPy
+    r"""Calculate precipitable water through the depth of a sounding.
+
+    Default layer depth is sfc-400 hPa. Formula used is from the AMS glossary:
+
+    \begin{align}
+    \frac{1}{pg} \int\limits_0^d x \,dp
+    \end{align}
+
+    Parameters
+    ----------
+    dewpt : array-like
+        Atmospheric dewpoint profile
+    p : array-like
+        Atmospheric pressure profile
+    top: `pint.Quantity`
+        The top of the layer, specified in pressure.
+
+    Returns
+    -------
+    `pint.Quantity'
+        The precipitable water in the layer, in inches
+
+    """
+    pw_layer = get_layer(p, dewpt, depth=p[0] - top)
+
+    w = mixing_ratio(saturation_vapor_pressure(pw_layer[1]), pw_layer[0])
+
+    pw = (((w[:-1].magnitude + w[1:].magnitude) / 2.) *
```

Comment:
Instead of doing this manually, try using `numpy.trapz()`. I'd expect it to read something like:
```python
pw = np.trapz(w, p_layer) / (g * rho_l)
```